### DISCUSSION TITLE
Make SWT a standalone project

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/eclipse.platform.swt#1621] [Ideas]
SWT is currently part of eclipse-platform but:

- In contrast to other platform projects it does not allowed to depend on anything from platform (e.g. recently OSGi or Workspace/Resources)
- As it can be used in other context, it should/could (until today) not depend on any dependency (e.g. not using anything from the sdk-target platform.
- Is quite standalone (just a single jar + some fragments)
- Does not has "internal" API (except some platform specific fields)

Therefore SWT is more an own framework/library and could benefit from independent development cycles and seems perfectly suited to be an own project. This does not mean platform can't build/use against "latest" SWT ... it just would mean SWT *can* be released / developed independently.

I therefore like to suggest to split SWT from eclipse-platform, a first step could be to simply move the repository to its own organization (thanks to github redirection this would be quite transparent to users).

**Comment by iloveeclipse** [comment]
> benefit from independent development cycles

What would be improved by that proposal? Who will run the development & sync with platform releases? Some new contributors or the same people who do not have time to work on SWT already?

Why not wait for a constant stream of patches for SWT that would require independent development cycles, and once we see the need we can discuss whether moving to a different organisation would improve the process?

**Reply by HeikoKlare** [reply]
I agree that SWT can be considered a standalone project, as it does not depend on anything else from Eclipse Platform but can (and is) used by other projects than Eclipse. But whether it should be managed as a standalone project is another question, which should be driven by some actual benefits. So in this regard, I can second Andrey's question:
> What would be improved by that proposal?

I do actually not see the benefit of independent development cycles, as for an Eclipse release you will still require to have a stable version of SWT, such that you still need to align releases with Eclipse and you may only have more frequent releases of SWT. But is that really necessary?
